spec/csv_generator/csv_generator_spec.rb

```diff
@@ -1,256 +1,226 @@
-      csv_string = csv.generate_sample_sheet
-      expect(csv_string.class).to eq String
-    end
-
-    it 'must have the correct headers' do
-      csv_string = csv.generate_sample_sheet
-      headers = CSV.parse(csv_string)[0]
-
-      expected_headers = Pipelines.pacbio.sample_sheet.columns.map(&:first)
-      expect(headers).to eq(expected_headers)
-    end
-
-    it 'must have the correct well header rows' do
-      csv_string = csv.generate_sample_sheet
-      array_of_rows = CSV.parse(csv_string)
-
-      well_data_1 = array_of_rows[1]
-      well_data_2 = array_of_rows[2]
-
-      expect(well_data_1).to eq([
-        well1.plate.run.system_name,
-        well1.plate.run.name,
-        'true',
-        well1.position,
-        well1.pool_barcode,
-        well1.movie_time.to_s,
-        well1.insert_size.to_s,
-        well1.template_prep_kit_box_barcode,
-        well1.binding_kit_box_barcode,
-        well1.plate.run.sequencing_kit_box_barcode,
-        well1.on_plate_loading_concentration.to_s,
-        well1.plate.run.dna_control_complex_box_barcode,
-        well1.plate.run.comments,
-        well1.all_libraries_tagged.to_s,
-        '',
-        well1.barcode_set,
-        well1.same_barcodes_on_both_ends_of_sequence.to_s,
-        well1.find_sample_name,
-        well1.automation_parameters,
-        well1.generate_hifi,
-        well1.ccs_analysis_output
-      ])
-
-      expect(well_data_2).to eq([
-        well2.plate.run.system_name,
-        well2.plate.run.name,
-        'true',
-        well2.position,
-        well2.pool_barcode,
-        well2.movie_time.to_s,
-        well2.insert_size.to_s,
-        well2.template_prep_kit_box_barcode,
-        well2.binding_kit_box_barcode,
-        well2.plate.run.sequencing_kit_box_barcode,
-        well2.on_plate_loading_concentration.to_s,
-        well2.plate.run.dna_control_complex_box_barcode,
-        well2.plate.run.comments,
-        well2.all_libraries_tagged.to_s,
-        '',
-        well2.barcode_set,
-        well2.same_barcodes_on_both_ends_of_sequence.to_s,
-        well2.find_sample_name,
-        well2.automation_parameters,
-        well2.generate_hifi,
-        well2.ccs_analysis_output
-      ])
-    end
-
-    it 'must not have sample rows' do
-      csv_string = csv.generate_sample_sheet
-      array_of_rows = CSV.parse(csv_string)
-
-      expect(array_of_rows.size).to eq 3
-    end
-  end
-
-  context '#sort wells method' do
-    it '#sorts the wells by column' do
-      sorted_wells = []
-        for column in '1'...'12'
-          for row in 'A'...'H' do
-            sorted_wells.push(create(:pacbio_well, row: row, column: column))
-          end
-        end
-      plate = create(:pacbio_plate, wells: sorted_wells) 
-      run = create(:pacbio_run, plate: plate)
-      csv = ::CsvGenerator.new(run: run, configuration: Pipelines.pacbio.sample_sheet)
-
-      expect(csv.send(:sort_wells, sorted_wells.shuffle)).to eq(sorted_wells)
-    end
-  end
```

Comment:
Stopped testing the private method, instead test the resulting public behaviour.